compiler/vm/vmcompilerserdes.nim

```diff
@@ -212,6 +212,16 @@ proc deserializeArray*(
   if hasError:
     result = c.config.wrapError(result)
 
+proc deserializeNimNode*(c: TCtx, n: PNode, formal: PType,
+                         info: TLineInfo): PNode =
+  ## Makes sure `n` (which represent a NimNode value) is valid and turns it
+  ## into a proper NimNode literal.
+  if cyclicTree(n):
+    result = c.config.newError(
+      newNodeIT(nkEmpty, info, formal),
+      PAstDiag(kind: adCyclicTree, cyclic: n))
+  else:
+    result = newTreeIT(nkNimNodeLit, info, formal): n
```

Comment:
Don't see this as an issue now, and probably ever, but it looks like we're going to treat `nkNimNodeLit` to mean either a literal/static AST value or dynamic AST value. Where the latter one is generated on the fly, at compile time.